returnn/tf/layers/rec.py

```diff
@@ -362,23 +362,22 @@ def get_out_data_from_opts(cls, network, unit, sources=(), initial_state=None, *
     """
     from tensorflow.python.util import nest
     source_data = get_concat_sources_data_template(sources) if sources else None
-    if source_data and not source_data.have_time_axis():
-      # We expect to be inside another RecLayer, and should do a single step (like RnnCellLayer).
-      out_time_dim_axis = None
-      out_batch_dim_axis = 0
-    else:
-      out_time_dim_axis = 0
-      out_batch_dim_axis = 1
+    has_time = False if source_data and not source_data.have_time_axis() else True
```

Comment:
This is very strange to read. Just directly write:
```suggestion
    has_time = not source_data or source_data.have_time_axis()
```